### PR TITLE
Translator: You can use fanyi.youdao.com as a translator

### PR DESCRIPTION
**Pull request by muxueqz**
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9040)

### CODE REVIEW

File: frontend/ui/translator.lua [hunk 1]
```diff
@@ -146,6 +146,7 @@ local Translator = {
     trans_servers = {
         "https://translate.googleapis.com/",
         -- "http://translate.google.cn",
+    "https://fanyi.youdao.com",
```
Comment (Frenzie) [inline frontend/ui/translator.lua:149]: 4 spaces for indentation please, not 2. Besides that I assume it's mostly just copied from the other one?

File: frontend/ui/translator.lua [hunk 2]
```diff
@@ -349,6 +351,122 @@ Returns decoded JSON table from translate server.
...
+        }
+      })
+    end
+  end
+  local google_result = {}
+  google_result[1] = translateResult
+  google_result[6] = google_result_6
+  logger.dbg("translator json fake google:", google_result)
+  return google_result
+end
```
Comment (Frenzie) [inline frontend/ui/translator.lua:459]: Probably a bit clearer to replace that too.

File: frontend/ui/translator.lua [hunk 3]
```diff
@@ -349,6 +351,122 @@ Returns decoded JSON table from translate server.
...
+    local sink = {}
+    if request == nil then
+        request = {}
+    end
+    socketutil:set_timeout()
+
+    request['url'] = s_url
+    request['method'] = method
+    request['sink'] = ltn12.sink.table(sink)
+    request['headers'] = headers
```
Comment (Frenzie) [inline frontend/ui/translator.lua:376]: Please use double quotes here as per the style guide. Besides that it looks good!

Comment (muxueqz) [reply frontend/ui/translator.lua]: Thanks, it's douban quotes now

File: frontend/ui/translator.lua [hunk 4]
```diff
@@ -349,6 +351,122 @@ Returns decoded JSON table from translate server.
 @treturn string result, or nil
 --]]
 function Translator:loadPage(text, target_lang, source_lang)
+    local trans_server = self:getTransServer()
+    if trans_server == "https://fanyi.youdao.com" then
+        return self:loadPageByYoudao(text, target_lang, source_lang)
+    end
+    return self:loadPageByGoogle(text, target_lang, source_lang)
```
[the remark this answers is not in the document]
Comment (muxueqz) [reply frontend/ui/translator.lua]: I just updated it , do you like it now ?

**Comment by muxueqz** [comment]
@Frenzie emm.... fanyi.youdao is not support dutch to english, so this test is error

**Comment by Frenzie** [comment]
> emm.... fanyi.youdao.com is not support dutch to english, so this test is error

That cannot be the reason. See https://github.com/koreader/koreader/blob/f37699a12b893984690f0e433b7f048af3c621a7/spec/unit/translator_spec.lua

**Comment by muxueqz** [comment]
> > emm.... fanyi.youdao.com is not support dutch to english, so this test is error
> 
> That cannot be the reason. See https://github.com/koreader/koreader/blob/f37699a12b893984690f0e433b7f048af3c621a7/spec/unit/translator_spec.lua

Yes. I got wrong before, that seems a Lua module problem?
self was not pass to self:loadPageByGoogle, so self:loadPageByGoogle's self.trans_params is nil
Do you have any suggestions ?

**Comment by Frenzie** [comment]
Those are fair points. (Sorry, I didn't have time to look closer, but I think I preferred the previous state of the PR. :-) )

Google is definitely quite good at Chinese to English; I assume the reverse is true as well.

**Comment by poire-z** [comment]
Closing because lack of answers/activity.

**Pull request closed without merging**